NFT/src/pages/api/files/[filter].ts

```diff
@@ -0,0 +1,22 @@
+import pinataSDK, { PinataMetadataFilter } from "@pinata/sdk";
+import { NextApiRequest, NextApiResponse } from "next";
+
+const pinata = new pinataSDK({ pinataJWTKey: process.env.PINATA_JWT });
+
+export const getNFTMetadata  = async (filter?: PinataMetadataFilter) => {
+  const metadata = filter;
```

Comment:
The filter is an option field though we are querying it as a string below `const metadata = JSON.parse(req.query.filter as string);`, should consider adding a check in case of unexpected value or an empty string.